### PR TITLE
handle username validation errors states better

### PR DESCRIPTION
**Pull request by benjiwheeler**
### Resolves:

Step towards resolving #3053

### Changes:

* Focusing on a field clears the error (until field is blurred again)
* If username and password are the same, password field reports error right away (instead of waiting for user to click "Next")
* Design of inputs with an error that are focused on are no longer a weird combination of blue and orange borders!

### Screenshots

Focusing on a field clears the error (until field is blurred again):

clearerror

If username and password are the same, password field reports error right away (instead of waiting for user to click "Next"):

errorrightaway

### CODE REVIEW

File: src/lib/validate.js [hunk 1]
```diff
@@ -40,13 +40,21 @@ module.exports.validateUsernameRemotely = username => (
...
+ * @return {object}              {valid: boolean, errMsgId: string}
+ */
+module.exports.validatePassword = (password, username) => {
     if (!password) {
         return {valid: false, errMsgId: 'form.validationRequired'};
     } else if (password.length < 6) {
         return {valid: false, errMsgId: 'registration.validationPasswordLength'};
     } else if (password === 'password') {
         return {valid: false, errMsgId: 'registration.validationPasswordNotEquals'};
+    } else if (username && password === username) {
```
Comment (picklesrus) [inline src/lib/validate.js]: Can you add a test for this case?

File: src/components/formik-forms/formik-input.scss [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+        &:focus {
+            box-shadow: 0 0 0 .25rem $ui-orange-25percent;
+            outline: none;
+        }
+    }
+
+    /* IE10/11-specific style resets */
+    &::-ms-reveal, &::-ms-clear {
+        display: none;
```
[the remark this answers is not in the document]
Comment (benjiwheeler) [reply src/components/formik-forms/formik-input.scss]: I carried these over when I copied similar code. Ray suggests we try omitting them here, so i'll do that.

File: src/lib/validate.js (hunk 1 above)
Comment (benjiwheeler) [reply src/lib/validate.js]: Done

**Comment by benjiwheeler** [comment]
Updated this slightly, per feedback from Katny:
1) keep showing validation errors on focus, until first keystroke
2) prioritize vulgarity over other validations

File: src/components/join-flow/username-step.jsx [hunk 3]
```diff
@@ -36,20 +36,26 @@ class UsernameStep extends React.Component {
     // we allow username to be empty on blur, since you might not have typed anything yet
     validateUsernameIfPresent (username) {
         if (!username) return null; // skip validation if username is blank; null indicates valid
+        // if username is not blank, run both local and remote validations
         const localResult = validate.validateUsernameLocally(username);
-        if (localResult.valid) {
-            return validate.validateUsernameRemotely(username).then(
-                remoteResult => {
-                    if (remoteResult.valid) return null;
+        return validate.validateUsernameRemotely(username).then(
```
Comment (picklesrus) [inline src/components/join-flow/username-step.jsx]: hmm, but how aren't we making a network request all the time? I'm not sure this is a worthwhile tradeoff.

Comment (benjiwheeler) [reply src/components/join-flow/username-step.jsx]: Resolved as ok, per discussion in join flow subcommittee

**Pull request merged**